src/conv/heuristics/ai_heuristics.cpp

```diff
@@ -509,11 +515,16 @@ std::shared_ptr<Model> GetModel(const std::string& arch, const std::string& solv
 bool ModelSetParams(const std::string& arch,
                     const std::string& solver,
                     const std::vector<float>& features,
-                    std::function<bool(int, int)> validator)
+                    bool transform_features,
+                    std::function<bool(std::size_t, std::string)> validator)
 {
-    auto model             = GetModel(arch, solver);
-    int dim                = std::sqrt(features.size());
-    fdeep::tensors context = model->Encode(features, dim);
+    auto model = GetModel(arch, solver);
+    int dim    = 0;
+    if(transform_features)
+        dim = std::sqrt(features.size());
+    else
+        dim = features.size();
+    fdeep::tensors context = model->Encode(features, dim, transform_features);
```

Comment:
can we make `dim` a part of the `Model` object? Since `dim` is known apriori, maybe we can just add it to the metadata for the model.

Reply:
This would be fine but I don't see the point as we have to hard-code the feature vector anyway so if we know the size of the feature vector we can just extract ```dim``` from it.